src/main/java/com/limechain/rpc/methods/system/SystemRPCImpl.java

```diff
@@ -132,9 +145,19 @@ public void systemRemoveReservedPeer(String peerId) {
     /**
      * Returns the state of the syncing of the node.
      */
-    // TODO: Implement in Mx.
     public Map<String, Object> systemSyncState() {
-        return null;
+        long highestBlock = this.connectionManager
+                .getPeerIds()
+                .stream()
+                .map(this.connectionManager::getPeerInfo)
+                .mapToLong(peerInfo -> peerInfo.getBestBlock().longValue())
+                .max().orElse(0);
+
+        return Map.ofEntries(
+                entry("startingBlock", this.syncedState.getStartingBlockNumber()),
+                entry("currentBlock", this.syncedState.getLastFinalizedBlockNumber()),
+                entry("highestBlock", highestBlock)
+        );
```

Comment:
Are we sure highestBlock should be retrieved this way?

Reply:
This is the highestBlock we are aware of based on information provided by other peers by blockannaunce. Not sure which highest block to provide otherwise except for the lastfinalizedbock again